### DISCUSSION TITLE
Is there a way to add a filter and sort method inside the table ?

### DISCUSSION THREAD

**Discussion by thibaudbrault** [petyosi/react-virtuoso#600] [Q&A] [answered]
I'm currently using virtuoso for rendering a table and before that I was using vanilla React JS.  
I have a search field that filter the rows in the table with the text in the input and the rows were ordered in a descending order by the name returned in a <td>.  
I had this code before the <tr> that I want to add : 
````
{abilities.filter((abilities) => {
    if (search === "") {
        return abilities
    } else if (abilities.name.replace(/-/g, ' ').toLowerCase().includes(search.toLowerCase())) {
        return abilities
    }
})
.sort((a, b) => a.name.localeCompare(b.name))
````
Is it possible to add it ? If not is there something equivalent for a search field and for the sorting ?  
Also, is it possible to give a classname to the table tag when using TableVirtuoso ?

**Comment by wiktoriavh** [comment, marked as answer]
inside your Virtuoso table you have the prop `data={yourData}`. In an earlier call you can manipulate `yourData` and pass that to virtuoso, instead of manipulating it inside of it.

```js
function YourTable() {
  const data = /* where you get your data from */;
  const [yourData, setYourData] = useState(data);

 function filterData() {
    // your manipulation
  setYourData(/* your result fo manipulation */)
  }

  return <VirtuosoTable
                data={yourData}
                /* other code */
              />
}
```

> Also, is it possible to give a classname to the table tag when using TableVirtuoso ?

you can make a custom table element by passing it into the components
```js
function YourTable() {
  return <VirtuosoTable
                components={{
                  Table: (props) => /* <our custom table here */
                
                /* other code */
              />
}
```
